nixos/modules/services/networking/unbound.nix

```diff
@@ -6,41 +6,116 @@ let
+    validSettingsTypes = with types; oneOf [ validSettingsPrimitiveTypes (listOf validSettingsPrimitiveTypes) ];
+    settingsType = with types; attrsOf validSettingsTypes
+      // { description = ''
+            unbound.conf configuration type. The format consist of an attribute
+            set of settings. Each settings can be either one value of a list of
+            values. The allowed values are integers, strings, booleans or
+            floats.
+           '';
+        };
+  in {
+    options = {
+      server = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          server: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      remote-control = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          remote-control: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      stub-zone = mkOption {
+        type = with types; listOf settingsType;
+        default = [ ];
+        description = ''
+          stub-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      forward-zone = mkOption {
+        type = with types; listOf settingsType;
+        default = [ ];
+        example = literalExample ''
+          [
+            {
+              name = ".";
+              forward-addr = "1.1.1.1@853#cloudflare-dns.com";
+            }
+            {
+              name = "example.org.";
+              forward-addr = [
+                "1.1.1.1@853#cloudflare-dns.com"
+                "1.0.0.1@853#cloudflare-dns.com"
+              ];
+            }
+          ];
+        '';
+        description = ''
+          forward-zone: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+      python = mkOption {
+        type = settingsType;
+        default = { };
+        description = ''
+          python: clause configuration.
+          See the <citerefentry><refentrytitle>unbound.conf</refentrytitle>
+          <manvolnum>5</manvolnum></citerefentry> manpage for a list of
+          available options.
+        '';
+      };
+    };
+  };
+
+  yesOrNo = v: if v then "yes" else "no";
+
+  toOption = n: v: "${toString n}: ${v}";
+
+  toConf = n: v:
+    if builtins.isFloat v then (toOption n (toString v))
```

Comment:
Any precision concerns here?

Reply:
This will most likely produce strings with exceedingly long precision but should be ok. To avoid this `builtins.toJSON` could be used instead. See
```
nix-repl> toString 0.3        
"0.300000"

nix-repl> builtins.toJSON 0.3
"0.3"
```